src/plugins/dashboard/public/dashboard_top_nav/internal_dashboard_top_nav.tsx

```diff
@@ -153,6 +154,11 @@ export function InternalDashboardTopNav({
     title,
   ]);
 
+  /** Set chrome tab title when dashboard's title changes */
+  useEffect(() => {
+    docTitle.change(title);
+  }, [title, docTitle]);
```

Comment:
I do not want to change the `docTitle` for portable dashboards - so, I do not want this `useEffect` to be part of the `DashboardRenderer` component. This means the best place for it is in the `DashboardTopNav`, since this will **only** happen in the Dashboard app specifically.

Reply:
I think the Security app also uses this component. Does this work in that context as well?

Reply:
Oh, good catch - you're right 🤦 Looks like they do - and the browser title is being changed:

![image](https://github.com/elastic/kibana/assets/8698078/24918b40-f904-4dc5-a32d-0fd0f2ad57be)

Do you think this is okay?

Reply:
My guess is that this is alright, but @angorayc should maybe weigh in here.

If the security solution doesn't want the tab titles to reflect the Dashboard title, we could maybe move this `useEffect` onto the `dashboard_app` side of things.

Reply:
> If the security solution doesn't want the tab titles to reflect the Dashboard title, we could maybe move this `useEffect` onto the `dashboard_app` side of things.

Yeah, I thought about that - unfortunately it's not as simple, since we are outside the dashboard context and so it's not as easy to respond to title changes. But I'm sure I could find a way around that if that's the way we want to go :)

Reply:
Yeah it isn't extremely clean, but we do have the Dashboard's external API to use in this case. The ugly part is that it requires a separate component. It could be done like:

1: create a separate component (I just put it at the top of `dashboard_app`)
``` typescript
const DashboardTabTitleSetter = ({ dashboardAPI }: { dashboardAPI: DashboardAPI }) => {
  const {
    chrome: { docTitle },
  } = pluginServices.getServices();
  const title = dashboardAPI.select((state) => state.explicitInput.title);
  useEffect(() => {
    docTitle.change(title);
  }, [title, docTitle]);
  return null;
};
```

2: Render this above the `TopNav` on line 213
``` typescript
{dashboardAPI && (
            <>
              <DashboardTabTitleSetter dashboardAPI={dashboardAPI} />
              <DashboardTopNav
                redirectTo={redirectTo}
                embedSettings={embedSettings}
                dashboardContainer={dashboardAPI}
              />
            </>
          )}
```

Reply:
We'll see if @angorayc is okay with the title changing - if that is undesired behaviour for the Security context, I can definitely go the separate component route :+1:

Reply:
Hi @Heenawter @ThomThomson , thanks so much for considering the use case for SecuritySolution!
But yeah, Security Solution would prefer not to override the document title with dashboard title, as we have an existing logic for displaying the dashboard title.

Reply:
@angorayc Yup, makes total sense!! :) I'll do what @ThomThomson suggested to prevent this from happening, then - thanks so much for the response! 🙇

Reply:
Fixed by fba6e3288b3a6debcd9e95af982e861c469ff4ed 🎉  The tab title changing now only applies to the Dashboard app specifically - the Security dashboards are no longer impacted.


Uploading Screen Recording 2023-11-16 at 3.06.16 PM.mov…